tests/mcaptcha/default.nix

```diff
@@ -0,0 +1,103 @@
+{modules, ...}: let
+  port = 7000;
+  urlRoot = "http://localhost:${builtins.toString port}";
+in {
+  name = "mcaptcha tests";
+
+  nodes.create_locally = {pkgs, ...}: {
+    imports = [
+      modules.default
+      modules.mcaptcha
+    ];
+
+    services.mcaptcha.enable = true;
+    services.mcaptcha.settings.server.port = port;
+    services.mcaptcha.server.cookieSecretFile = pkgs.writeText "cookie-secret" "mcaptcha-cookie-secret-dm0tdGVzdC1ydW4tbWNhcHRjaGEtdGVzdHM";
+    services.mcaptcha.captcha.saltFile = pkgs.writeText "salt" "asdl;kjfhjawehfpa;osdkjasdvjaksndfpoanjdfainsdfaijdsfajlkjdsaf;ajsdfweroire";
+    services.mcaptcha.database.createLocally = true;
+    services.mcaptcha.redis.createLocally = true;
+
+    # the following two lines are for debugging
+    networking.firewall.enable = false;
+    services.mcaptcha.settings.server.ip = "0.0.0.0";
+  };
+
+  nodes.bring_your_own_services = {
+    pkgs,
+    config,
+    ...
+  }: {
+    imports = [
+      modules.default
+      modules.mcaptcha
+    ];
+
+    services.mcaptcha.enable = true;
+    services.mcaptcha.settings.server.port = port;
+    services.mcaptcha.settings.server.domain = "localhost";
+    services.mcaptcha.server.cookieSecretFile = pkgs.writeText "cookie-secret" "mcaptcha-cookie-secret-dm0tdGVzdC1ydW4tbWNhcHRjaGEtdGVzdHM";
+    services.mcaptcha.captcha.saltFile = pkgs.writeText "salt" "asdl;kjfhjawehfpa;osdkjasdvjaksndfpoanjdfainsdfaijdsfajlkjdsaf;ajsdfweroire";
+
+    services.mcaptcha.settings.database.name = "my_mcaptcha";
+    services.mcaptcha.settings.database.username = "role_mcaptcha";
+    services.mcaptcha.settings.database.hostname = "my_own_services";
+    services.mcaptcha.settings.database.port = 5432;
+    services.mcaptcha.database.passwordFile = pkgs.writeText "db-password" "mcaptcha-db-secret";
+
+    services.mcaptcha.redis.passwordFile = pkgs.writeText "redis-secret" "(*&(*):ps@r}";
+    services.mcaptcha.redis.host = "my_own_services";
```

Comment:
Again, probably a matter of taste, but I don't like giving the full path to each option. My eyes have to work way too much. I prefer attribute sets.

Reply:
Sure, we don't mind, but there are so many of these we don't feel it's worth our time. How does that sound?

Reply:
You select the whole block (something like `va{` in Vim), and then you replace (something like `:%s/\s+services\.mcaptcha//` in Vim). Then you ensure that you have braces around the block. Then you format the file using `alejandra`. That takes less than a minute. If you have good LSP integration, then merging such declarations into an attribute set is even easier. Also, you have introduced these lines at some point. When you introduce the third or fourth line with this much redundancy, then it'd be a good time to think about how to get rid of it right there and then.

Reply:
Resolving in favor of [our previous comment](https://github.com/ngi-nix/ngipkgs/pull/61#discussion_r1369839408).